airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/AirbyteFileOffsetBackingStore.java

```diff
@@ -59,14 +60,30 @@ public Map<String, String> read() {
   public void persist(final JsonNode cdcState) {
     final Map<String, String> mapAsString =
         cdcState != null ? Jsons.object(cdcState, Map.class) : Collections.emptyMap();
-    final Map<ByteBuffer, ByteBuffer> mappedAsStrings = mapAsString.entrySet().stream().collect(Collectors.toMap(
+
+    final Map<String, String> updatedMap = updateStateForDebezium2_1(mapAsString);
```

Comment:
Can we just call this function if the state is not Debezium 2.1?

Reply:
I did avoid the mutation if its already 2.1 state as seen in the method but the problem is that I need to look into the contents of the map and do some logic on top of it to determine if it is. 
```

      if (i == 0 && i1 == key.length() - 1) {
        // The state is Debezium 2.1 compatible. No need to change anything.
        return mapAsString;
      }
```

Reply:
Ok... if you say so, for me it's not easy to read, something like this would be much easier:
```
map = get the map
if (isOldVersion()
  transformTo2-1
...
```
With the logic you have in-place, you are creating object that will not be used after all.